ec2-spot-elastic-inference/ec2-spot-elastic-inference.sh

```diff
@@ -0,0 +1,122 @@
+#!/bin/bash -x
+
+echo "Hello World from EC2 Spot Team..."
+#Global Defaults
+DEFAULT_REGION=us-east-1
+SLEEP=5
+LAUNCH_TEMPLATE_NAME=EC2SpotElasticInference-LT
+LAUNCH_TEMPLATE_VERSION=1
+CFS_STACK_NAME=EI-Spot-Stack
+CFS_STACK_FILE=EI_Spot_CFN.yaml
+declare -a CFK_STACK_OP_KEYS_LIST=("awsRegionId"  "eventRule" "instanceProfile" 
+                                    "instanceRole" "instanceSecurityGroup" "publicSubnet" 
+                                    "snsTopic" "vpc" "vpcEndpoint" )
+
+
+
+cp -Rfp templates/*.json .
+cp -Rfp templates/*txt .
+
+#CFS_STACK_ID=$(aws cloudformation create-stack --stack-name $CFS_STACK_NAME  --template-body file://$CFS_STACK_FILE --capabilities CAPABILITY_IAM --region $DEFAULT_REGION|jq -r '.StackId')
+echo "Created the stack $CFS_STACK_NAME with Stack Id $CFS_STACK_ID. Please wait till the status is COMPLETE"
+
+aws cloudformation wait stack-create-complete --stack-name $CFS_STACK_NAME --no-paginate
+
+
+for CFK_STACK_OP_KEY in "${CFK_STACK_OP_KEYS_LIST[@]}"; do
+    CFK_STACK_OP_VALUE=$(aws cloudformation describe-stacks --stack-name $CFS_STACK_NAME | jq -r ".Stacks[].Outputs[]| select(.OutputKey==\"$CFK_STACK_OP_KEY\")|.OutputValue")
+    echo "$CFK_STACK_OP_KEY=$CFK_STACK_OP_VALUE"
+    #sed -i.bak -e "s#%$CFK_STACK_OP_KEY%#$CFK_STACK_OP_VALUE#g"  user-data.txt
+    sed -i.bak -e "s#%$CFK_STACK_OP_KEY%#$CFK_STACK_OP_VALUE#g"  EI_launch_template.json
+    sed -i.bak -e "s#%$CFK_STACK_OP_KEY%#$CFK_STACK_OP_VALUE#g"  ec2_fleet.json
+    
+done
+
+export AMI_ID=$(aws ec2 describe-images --owners amazon --filters 'Name=name,Values=Deep Learning AMI (Amazon Linux 2) Version*' 'Name=state,Values=available' --output json | jq -r '.Images |   sort_by(.CreationDate) | last(.[]).ImageId')
+echo "Amazon AMI_ID is $AMI_ID"
+
+sed -i.bak  -e "s#%ami-id%#$AMI_ID#g"  EI_launch_template.json
+
+LAUCH_TEMPLATE_ID=$(aws ec2 create-launch-template --launch-template-name $LAUNCH_TEMPLATE_NAME --version-description LAUNCH_TEMPLATE_VERSION --launch-template-data file://EI_launch_template.json | jq -r '.LaunchTemplate.LaunchTemplateId')
+#LAUCH_TEMPLATE_ID=lt-0bc4d78f36f8d4efb
+echo "Amazon LAUCH_TEMPLATE_ID is $LAUCH_TEMPLATE_ID"
+
+FLEET_ID=$(aws ec2 create-fleet --cli-input-json file://ec2_fleet.json | jq -r '.FleetId')
+#FLEET_ID=fleet-a23a9a64-3f94-4fc2-8461-3d2c11b3f84e
+echo "Amazon LAUCH_TEMPLATE_ID is $FLEET_ID"
+
+aws ec2 describe-fleets --fleet-id $FLEET_ID
+
+
+exit 0
+
+sleep 5
+
+
+if [ "1" == "2" ]; then
```

Comment:
Why ? Not sure what this does, and why are we uploading it. We have an exit 0 above, and then an Sleep and then a condition that is null, This looks to me as we are uploading files we have been tinkering with and are half way, why? Are they truly needed?  Otherwise , back again we are making our life more difficult at curating and offering consumable knowledge to Spot users.